pandas/_libs/tslibs/strptime.pyx

```diff
@@ -20,10 +20,10 @@ from numpy cimport (
     ndarray,
 )
 
+from pandas._libs.missing cimport checknull_with_nat_and_na
```

Comment:
is there any way to do this that avoids dependencies outside of tslibs?

Reply:
I am not able to find any pd.NA in tslibs.
Is there any reason why we want to limit the dependency?

Reply:
Hi @jbrockmendel , made some change. Let me know if this works.

Reply:
> Is there any reason why we want to limit the dependency?

It isn't strictly necessary, but we've made an effort to make tslibs not depend on non-tslibs code to simplify the dependency structure.  If there isn't a viable alternative, its not _that_ big of a deal.

Reply:
yeah i am ok with this, though as @jbrockmendel notes this does some hoops in the dependency graph. though maybe this means that NA needs to be the lowest level.

cc @jbrockmendel

Reply:
> though maybe this means that NA needs to be the lowest level.

I've thought about this and prefer to punt on it for as long as possible to avoid dealing with e.g. pd.Timestamp(pd.NA)

For this PR, I think that casting pd.NA to pd.NaT is inconsistent with the idea of pd.NA being semantically distinct from NaT/nan.  Doing this correctly requires #40932.  Until then, my preference would be to raise and tell users to do `arr[arr.isna()] = NaT`

Reply:
well that proposal is a long ways off.
what is the issue with doing what is happening now? circular deps?

Reply:
> what is the issue with doing what is happening now?

Treating pd.NA as interchangeable with NaT is basically The Wrong Behavior, since pd.NA's meaning is specifically distinct from NaT.  If we start blurring that distinction in some places, we'll end up making things even less clear for users.

Better to raise with a useful exception message than do The Wrong Thing. 

This isnt a hill worth dying on for me.